### ISSUE TITLE
Make Mac OS open file associations optional

### ISSUE THREAD

**Issue by cjohansson** [onivim/oni#2093]
Hi! I tried out Oni and I think it's a interesting project, however I noticed that it is now my Mac OS default application for a lot of different file-types and I don't want that so now I need to manually change open file associations. Can you make that feature optional? I think it should be a opt-in feature to make Oni default application for different file-types.

**Comment by cjohansson** [comment]
Yeah I think VSCode has the same behavior as Oni but most big applications for macOS has it as an option-it feature, i.e. Google Chrome

**Comment by bryphe** [comment]
Thanks for the issue @cjohansson !  Yes, unfortunately from the DMG we don't have the option to turn this on or off (it's handled by the `plist`).

@Akin909 - I think the one option we'd have is to switch to a `pkg` installer - just means extra work getting that installer package set up though! Those tend to have a little bit more configuration available.

**Comment by cjohansson** [comment]
How about not doing anything at install but instead having a option in configuration like: 

* "use as default application for supported files" (boolean)
* "check if oni is default application for supported files at start-up" (boolean)

At first launch maybe have a confirmation box "Do you want to use Oni as your default application for supported files?" (boolean) and the answer to this sets configuration to option above?
